front/components/sparkle/AppLayout.tsx

```diff
@@ -427,6 +475,13 @@ export default function AppLayout({
             </div>
           </main>
         </div>
+
+        <div>
+          <ToggleSideBarButton
+            navigationBarOpen={navigationBarOpen}
+            setNavigationBarOpen={setNavigationBarOpen}
```

Comment:
To keep the state management within the main function component.
```suggestion
            toggleNavigationBarVisibility={(isVisible) => setNavigationBarOpen(isVisible)}
```